crates/hc_bundle/src/packing/mod.rs

```diff
@@ -95,44 +103,7 @@ pub async fn pack<M: Manifest>(
     };
     bundle.write_to_file(&target_path).await?;
     if serialize_wasm {
-        let target_path_folder = target_path
-            .parent()
-            .expect("target_path should have a parent folder");
-        let _write_serialized_result =
-            futures::future::join_all(bundle.bundled_resources().iter().map(
-                |(relative_path, bytes)| async move {
-                    // only pre-serialize wasm resources
-                    if relative_path.extension() == Some(std::ffi::OsStr::new("wasm")) {
-                        let ios_folder_path = target_path_folder.join("ios");
-                        let mut resource_path_adjoined = ios_folder_path.join(
-                            relative_path
-                                .file_name()
-                                .expect("wasm resource should have a filename"),
-                        );
-                        // see this code for rationale
-                        // https://github.com/wasmerio/wasmer/blob/447c2e3a152438db67be9ef649327fabcad6f5b8/lib/engine-dylib/src/artifact.rs#L722-L756
-                        resource_path_adjoined.set_extension("dylib");
-                        ffs::create_dir_all(ios_folder_path).await?;
-                        ffs::write(&resource_path_adjoined, vec![].as_slice()).await?;
-                        let resource_path = ffs::canonicalize(resource_path_adjoined).await?;
-                        match build_ios_module(bytes.as_slice()) {
-                            Ok(module) => match module.serialize_to_file(resource_path.clone()) {
-                                Ok(()) => {
-                                    info!("wrote ios dylib to {:?}", resource_path);
-                                    Ok(())
-                                }
-                                Err(e) => Err(HcBundleError::SerializedModuleError(e)),
-                            },
-                            Err(e) => Err(HcBundleError::ModuleCompileError(e)),
-                        }
-                    } else {
-                        Ok(())
-                    }
-                },
-            ))
-            .await
-            .into_iter()
-            .collect::<Result<Vec<_>, _>>()?;
+        build_preserialized_wasm(&target_path, &bundle).await?;
```

Comment:
move implementation behind feature flag -- only supported in wasmer_sys

Reply:
I see the `wamr` implementation panics unimplemented... I guess I don't really understand what preserializing is and when it is used. Is this going to cause any dev UX problems? Or not, since the dev nix environment includes only the _sys version so this will never come up?

Reply:
Yeah so this came from this PR: https://github.com/holochain/holochain/pull/2218/files

The function name is not very clear. It's pre-compiling and pre-serializing the wasm so that it can be included in your app bundle that gets code signed. The original intention was to enable building for ios and workaround the restriction on JIT compiled code. I think this is more generally useful than just for ios (i.e. to avoid the slow wasm compilation time on android app's first startup). It's a fine alternative to using the interpreter mode. I'd like to rename the functions and modify the docs to remove references to ios (in another pr).

So because it's using the wasmer compiler it won't work in wamr mode. I think I'm now realizing that the --dylib-ios flag should just not be available when using the wamr feature flag, and that whole code path should just be excluded. I'll update this to reflect.

Reply:
> So because it's using the wasmer compiler it won't work in wamr mode. I think I'm now realizing that the --dylib-ios flag should just not be available when using the wamr feature flag, and that whole code path should just be excluded. I'll update this to reflect.

Actually I'm not sure the best approach here. What do you think? Should the cli flag `--dylib-ios` flag not be available in wamr mode? Or should it be available and panic?